File: solution/array/058-length-of-last-word/solution.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLastWord = function (s) {
    let index = s.length - 1;

    // Skip trailing spaces
    while (index >= 0 && s[index] === ' ') {
        index--;
    }

    let length = 0;

    // Count characters of the last word
    while (index >= 0 && s[index] !== ' ') {
        length++;
        index--;
    }

    return length;
};

File: solution/array/058-length-of-last-word/solution.go
package main

func lengthOfLastWord(s string) int {
	n := len(s)
	index := n - 1

	// Skip trailing spaces
	for index >= 0 && s[index] == ' ' {
		index--
	}

	length := 0

	// Count characters of the last word
	for index >= 0 && s[index] != ' ' {
		length++
		index--
	}

	return length
}